### ISSUE TITLE
Add `GPT2Preprocessor`

### ISSUE THREAD

**Issue by chenmoneygithub** [keras-team/keras-nlp#682]
We need `GPT2Preprocessor` to do the following:
- tokenize inputs.
- Append "<|endoftext|>" to the beginning and end.

Notice that we should enforce the input to have one segment, since GPT2 naturally performs terribly on multi-segment classification tasks such as glue/mnli.

**Comment by jbischof** [comment]
For multi-segment there's also no canonical way to do this with GPT correct? Even prompts are part of the same segment?

**Comment by chenmoneygithub** [comment]
@jbischof Yes, everything is just one segment for GPT2.

**Comment by chenmoneygithub** [comment]
I realized it is a bit strange to pad "<|endoftext|>" to the beginning and end, because in the dataset the last token may not mark the end of a sentence/segment/paragraph. We may want to offer such start/end padding as optional, because for certain finetuning they are still useful.

**Comment by mattdangerw** [comment]
This is done https://keras.io/api/keras_nlp/models/gpt2/gpt2_preprocessor/
